[RnD-EasyPurchase-iOS][477785]Utilize Apple's StoreKit framework to handle the presentation of subscription options to users

shamin26 opened the pull request:
- Implemented a struct to represent the result of a product retrieval operation.
- Implemented a method to create and initiate a product retrieval request
- Implemented a method to fetch product information for a set of product IDs using necessaray callback responses.

Ref: https://redmine.bjitgroup.com/redmine/issues/477785

shamin26 committed:
[RnD-EasyPurchase-iOS][477785]Utilize Apple's StoreKit framework to handle the presentation of subscription options to users

- Implemented a struct to represent the result of a product retrieval operation.
- Implemented a method to create and initiate a product retrieval request
- Implemented a method to fetch product information for a set of product IDs using necessaray callback responses.

Ref: https://redmine.bjitgroup.com/redmine/issues/477785

shamin26 committed:
[RnD-EasyPurchase-iOS][477785]Utilize Apple's StoreKit framework to handle the presentation of subscription options to users

- Implemented a struct to represent the result of a product retrieval operation.
- Implemented a method to create and initiate a product retrieval request
- Implemented a method to fetch product information for a set of product IDs using necessaray callback responses.

Ref: https://redmine.bjitgroup.com/redmine/issues/477785

shamin26 committed:
[RnD-EasyPurchase-iOS][477785]Utilize Apple's StoreKit framework to handle the presentation of subscription options to users

- Implemented a struct to represent the result of a product retrieval operation.
- Implemented a method to create and initiate a product retrieval request
- Implemented a method to fetch product information for a set of product IDs using necessaray callback responses.

Ref: https://redmine.bjitgroup.com/redmine/issues/477785

RejaulHasan commented on Sources/EasyPurchase/ProductInfoController.swift:
```diff
@@ -0,0 +1,36 @@
+// A class that conforms to the `ProductsInfoHandler` protocol.
+class ProductInfoController: NSObject, ProductsInfoHandler {
+    // An instance of `InAppProductFetcherBuilder` used to build product retrieval requests.
+    var fetchProductBuilder = InAppProductFetcherBuilder()
```
Make it private if not needed by others.

shamin26 replied:
Done. Thank you.

RejaulHasan commented on Sources/EasyPurchase/ProductInfoController.swift:
```diff
@@ -0,0 +1,36 @@
+}
+
+// A class that conforms to the `ProductsInfoHandler` protocol.
+class ProductInfoController: NSObject, ProductsInfoHandler {
```
make class final it not need to inherit.

shamin26 replied:
Done. Thank you.

RejaulHasan commented on Sources/EasyPurchase/ProductInfoController.swift:
```diff
@@ -0,0 +1,36 @@
+import StoreKit
+
+// A protocol for handling product information retrieval.
+protocol ProductsInfoHandler {
```
1. I do not think your code may accessible by outside module. Have a look on object scope topic. It will help you to understand why it will happen
2. If this protocol only confirmed by apple then make it a class or anyobject type.

shamin26 replied:
Made it to AnyObject type. Thank you.

RejaulHasan reviewed (changes requested):
I do not think your code may accessible by outside module. Have a look on object scope topic. It will help you to understand why it will happen

RejaulHasan reviewed (commented):
We may update it time to time. Nice work.

shamin26 committed:
[IAP][Func][Review-Feedback]: Utilize Apple's StoreKit framework to handle the presentation of subscription options to users

- Fixed the class scope issue
- Fixed the protocol scope issue
- Fixed the variable scope issue

TulonBJIT reviewed (approved):
It looks ok on my side.

TulonBJIT committed:
Merge branch 'develop' of ssh://github.com/TulonBJIT/EasyPurchase into Task_477785_3
